support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```diff
@@ -304,6 +290,24 @@ const StripeForm = (props: StripeFormPropTypes) => {
     }
   }, [setupIntentClientSecret]);
 
+  // Handle the payment once all state updates to the card data, errors list, etc
+  // have completed after hitting the submit button
+  useEffect(() => {
+    if (stripe && submitReady && props.allErrors.length === 0 && cardErrors.length === 0 && !recaptchaError) {
```

Comment:
it's not clear what "ready" means here, perhaps `submitReady` could be `formSubmitted` or `submitPressed`?

Reply:
I think because we can set this `submitReady` value to `false` if the form has validation errors, `formSubmitted` or `submitPressed` may be misleading as setting those variable names to `false` implies the user hasn't clicked submit? 

How about `readyToSubmit`?

Reply:
`submitPressedAndValidationPending`? I have nothing against long names if they're clear.  It's just the word ready in my mind that is unclear. But happy to go with the majority view as scala conventions aren't the same as client side ones.

Reply:
I'm all for clearer variable names and I wasn't super happy with `submitReady`- switching it to `readyToSubmitPendingValidation`, which I think is about as clear as it can get.